cpp/examples/tpch/datagen/datagen.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -e
+
+scale_factor=$1
+pwd=$(pwd)
+
+# Clone the datafusion repo
+# This clone of the datafusion repo is uses a single thread
+# so that a single parquet file is generated for every table\
+if [ ! -d "datafusion" ]; then
+    git clone https://github.com/JayjeetAtGithub/datafusion.git datafusion
```

Comment:
will this break if any changes happen to the branch ? Is it better to use commit hash or tags, so that it remains consistent.

Reply:
Are there any commits in your fork that's needed for this example and  prevents us from using apache datafusion repo?

Reply:
Yeah, there is a commit that changes the file generation to single threaded so that we generate a single file. Can we use that change as a patch and use the datafusion repo ?

Reply:
I ended up doing the above, adding a patch for upstream datafusion. Please let me know if that looks good.